frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -24,69 +34,119 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
 }) => {
   const { t } = useTranslation();
   const [handlePromise, inProgress, errorMessage] = usePromiseHandler();
-  const submit = React.useCallback(
-    (event: React.FormEvent<HTMLFormElement>): void => {
-      event.preventDefault();
-      const deleteOptions = {
-        kind: 'DeleteOptions',
-        apiVersion: 'v1',
-        propagationPolicy: 'Foreground',
-      };
-      handlePromise(
-        Promise.all([
-          k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
-          ...(subscription?.status?.installedCSV
-            ? [
-                k8sKill(
-                  ClusterServiceVersionModel,
-                  {
-                    metadata: {
-                      name: subscription.status.installedCSV,
-                      namespace: subscription.metadata.namespace,
-                    },
-                  },
-                  {},
-                  deleteOptions,
-                ).catch(() => Promise.resolve()),
-              ]
-            : []),
-        ]),
-      )
-        .then(() => {
-          close();
-          if (
-            window.location.pathname.split('/').includes(subscription.metadata.name) ||
-            window.location.pathname.split('/').includes(subscription.status.installedCSV)
-          ) {
-            history.push(
-              resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()),
-            );
-          }
-        })
-        .catch(() => {});
-    },
-    [close, handlePromise, k8sKill, subscription],
+  const [consoleOperatorConfig] = useK8sWatchResource<K8sResourceKind>({
+    kind: referenceForModel(ConsoleOperatorConfigModel),
+    isList: false,
+    name: CONSOLE_OPERATOR_CONFIG_NAME,
+  });
+
+  const canPatchConsoleOperatorConfig = useAccessReview({
+    group: ConsoleOperatorConfigModel.apiGroup,
+    resource: ConsoleOperatorConfigModel.plural,
+    verb: 'patch',
+    name: CONSOLE_OPERATOR_CONFIG_NAME,
+  });
+
+  const csvPlugins = getClusterServiceVersionPlugins(csv?.metadata?.annotations);
+  const enabledPlugins = csvPlugins.filter((plugin) =>
+    isPluginEnabled(consoleOperatorConfig, plugin),
   );
 
+  const getPlugins = () => {
```

Comment:
We should make it clear this isn't returning an array of plugin names but text we show in the UI.

```suggestion
  const getPluginsLabel = () => {
```